core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -359,11 +359,38 @@ sealed abstract class IO[+A] private () extends IOPlatform[A] {
    */
   def evalOn(ec: ExecutionContext): IO[A] = IO.EvalOn(this, ec)
 
+  /**
+   * Shifts the execution of the current IO to the specified [[java.util.concurrent.Executor]].
+   *
+   * @param executor
+   * @return
+   */
+  def evalOnExecutor(executor: Executor): IO[A] = {
+    require(executor != null, "Cannot pass undefined Executor as an argument")
+    executor match {
+      case ec: ExecutionContext =>
+        evalOn(ec: ExecutionContext)
+      case executor =>
+        IO.executionContext.flatMap { refEc =>
+          val newEc: ExecutionContext =
+            ExecutionContext.fromExecutor(executor, refEc.reportFailure)
+          evalOn(newEc)
+        }
+    }
+  }
```

Comment:
Instead of duplicating the implementation :)

```suggestion
  def evalOnExecutor(executor: Executor): IO[A] =
    IO.asyncForIO.evalOnExecutor(executor)
```